src/EnergyPlus/VariableSpeedCoils.cc

```diff
@@ -6105,7 +6111,9 @@ namespace VariableSpeedCoils {
             state.dataVariableSpeedCoils->VarSpeedCoil(DXCoilNum).SimFlag = true;
         }
 
-        if (CompressorOp == CompressorOperation::Off) {
+        if (CompressorOp == CompressorOperation::Off ||
+            (state.dataVariableSpeedCoils->VarSpeedCoil(DXCoilNum).RatedCapCoolTotal <= 0.0 &&
+             state.dataVariableSpeedCoils->VarSpeedCoil(DXCoilNum).RatedCapCoolTotal != DataSizing::AutoSize)) {
```

Comment:
These 2 places (here and 7399) are in the Calc function so either SizeVarSpeedCoil has been called or it hasn't. In either case if <= 0.0 should be sufficient, no need to check for AutoSize each iteration of the simulation, just return and wait for sizing to complete.

Reply:
@rraustad Yes, but...not having the AutoSize case here caused a couple of unit test fails.  See the second to last commit where I put these in and then the unit tests went back to working again.

Reply:
I think the unit test did not set RatedCapCoolTotal? or it was set to AutoSize and sizing had not been called. It would be better to update the unit test than add this conditional here.

Reply:
@rraustad Ok, correct--in one unit test, this was not set.  So, I set it and was able to remove the AutoSize part of the condition in in 6114.  For the other unit test which was a HEATING coil test, I fixed this by removing the extra IF conditions in 7399.  They didn't make any sense there anyway because why look at the cooling capacity for a heating coil.